### ISSUE TITLE
SegmentationModelsBuilder default activation="softmax" maps to nn.Softmax() with no dim (implicit-dimension UserWarning on every call)

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4310]
## What happens

`kornia/models/segmentation/segmentation_models.py` (`SegmentationModelsBuilder.build`) defaults to `activation="softmax"` and passes it through to `segmentation_models_pytorch`, which maps that string to `nn.Softmax()` with no `dim`. PyTorch's implicit-dimension heuristic resolves it to `dim=1` for 4-D output, which is the intended per-pixel class softmax, but every forward emits

```
UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
```

so any user of `kornia[segmentation]` (extra added in #4301) gets a deprecation warning on every call, and the behaviour depends on a heuristic PyTorch has said it will remove.

Nothing in CI sees it today: `segmentation_models_pytorch` is installed nowhere in the matrix, and `tests/models/test_segmentation_models.py` is `importorskip`-guarded.

## Fix

Pass `activation="softmax2d"` (smp's name for `nn.Softmax(dim=1)`) as the default, or build the head with an explicit `nn.Softmax(dim=1)` after `smp.create_model(..., activation=None)`. Add a `pytest.warns`-free assertion to the existing smoke test so the warning cannot come back once the extra is installed somewhere.

Found during the review of #4301.

Posted on behalf of @ducha-aiki by Claude (Claude Fable 5.1).

**Comment by ducha-aiki** [comment]
This is moot after the rework of #4301 (head `064b334f`): `SegmentationModelsBuilder.build()` no longer imports `segmentation_models_pytorch` or passes any `activation` (or `model_name`/`encoder_name`/...) to it. It now takes a network the user has already constructed plus the encoder's preprocessing-parameter dict, so the softmax head, and whether it is `nn.Softmax(dim=1)`, `"softmax2d"` or nothing, is the user's own `smp.Unet(..., activation=...)` call; kornia has no default to fix and no smp call site left to emit the warning. The docs example uses `activation="softmax2d"` so a copy-paste gets the explicit-dim head. The PR carries `Closes #4310`.

Posted on behalf of @ducha-aiki by Claude (Claude Fable 5.1).
